File: backend/src/main/java/io/github/daegwonkim/backend/entity/UserNeighborhood.java
package io.github.daegwonkim.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;

import java.math.BigDecimal;
import java.util.UUID;

@Getter
@Entity
@Table(name = "user_neighborhoods")
public class UserNeighborhood {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "user_id", columnDefinition = "UUID", nullable = false)
    private UUID userId;

    @Column(name = "eupmyeondong_id", columnDefinition = "UUID", nullable = false)
    private UUID eupmyeondongId;

    @Column(precision = 10, scale = 8, nullable = false)
    private BigDecimal latitude;

    @Column(precision = 11, scale = 8, nullable = false)
    private BigDecimal longitude;
}

File: backend/src/main/kotlin/io/github/daegwonkim/backend/entity/UserNeighborhood.kt
package io.github.daegwonkim.backend.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.locationtech.jts.geom.Point
import java.util.UUID

@Entity
@Table(name = "user_neighborhoods")
class UserNeighborhood(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    val id: UUID? = null,

    @Column(name = "user_id", columnDefinition = "UUID", nullable = false)
    val userId: UUID,

    @Column(name = "neighborhood_id", columnDefinition = "UUID", nullable = false)
    val neighborhoodId: UUID,

    @Column(columnDefinition = "geometry(Point, 4326)")
    val location: Point
)